container-engine-lib/lib/backend_impls/docker/object_attributes_provider/object_attributes_provider.go

```diff
@@ -137,14 +138,33 @@ func (provider *dockerObjectAttributesProviderImpl) ForLogsStorageVolume() (Dock
 	return objectAttributes, nil
 }
 
-func (provider *dockerObjectAttributesProviderImpl) ForReverseProxy() (DockerObjectAttributes, error) {
-	name, err := docker_object_name.CreateNewDockerObjectName(reverseProxyName)
+func (provider *dockerObjectAttributesProviderImpl) ForReverseProxy(engineGuid engine.EngineGUID) (DockerObjectAttributes, error) {
+
+	nameStr := strings.Join(
+		[]string{
+			reverseProxyNamePrefix,
+			string(engineGuid),
+		},
+		objectNameElementSeparator,
+	)
+	name, err := docker_object_name.CreateNewDockerObjectName(nameStr)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred creating a Docker object name object from string '%v'", nameStr)
+	}
+
+	idLabelValue, err := docker_label_value.CreateNewDockerLabelValue(string(engineGuid))
 	if err != nil {
-		return nil, stacktrace.Propagate(err, "An error occurred creating a Docker object name object from string '%v'", reverseProxyName)
+		return nil, stacktrace.Propagate(err, "An error occurred creating the reverse proxy GUID Docker label from string '%v'", engineGuid)
+	}
+	guidLabelValue, err := docker_label_value.CreateNewDockerLabelValue(string(engineGuid))
```

Comment:
I'm not too familiar/don't remember what the implementation of `CreateNewDockerLabelValue` is but what's the difference between the `guid` and the `id` label value? Or are they meant to be the same.

Reply:
For some objects like services, the id is the name and the guid is the uuid.  For others like the engine, both id and guid labels are set to the uuid.  I did the same for the reverse proxy for consistency.

Reply:
ah yes I remember this now makes sense